### DISCUSSION TITLE
Support for unix sockets.

### DISCUSSION THREAD

**Discussion by marknefedov** [trillium-rs/trillium#82] [Ideas]
Unix socket support is usefull when app will be deployed behind a proxy (like nginx).
As an example warp currently allows to use tokio's UnixListener.
```rust
#[cfg(unix)]
#[tokio::main]
async fn main() {
    use tokio::net::UnixListener;
    use tokio_stream::wrappers::UnixListenerStream;

    pretty_env_logger::init();

    let listener = UnixListener::bind("/tmp/warp.sock").unwrap();
    let incoming = UnixListenerStream::new(listener);
    warp::serve(warp::fs::dir("examples/dir"))
        .run_incoming(incoming)
        .await;
}

#[cfg(not(unix))]
#[tokio::main]
async fn main() {
    panic!("Must run under Unix-like platform!");
}
```

**Comment by jbr** [comment]
This should be fairly easy to add, I just need to figure out the right logic for configuration. Are you familiar with a standardized env var for a socket path? I think ideally the server would allow explicit compiled-in configuration like `trillium_{runtime}::config().with_unix_socket("/var/run/trillium.sock").run(…)` but if that's not set (such as when someone's running with `trillium_{runtime}::run(…)`), it would check the environment for `SOCKET_PATH` or similar and use that if it's available, falling back to the current logic if not

**Reply by marknefedov** [reply]
I'm not sure if there is standardized environment variable. In most cases i've dealt with, it was app specific APP_CONF_SOCKET or something among those lines since you can have plenty of apps running at the same time with common platform on the same host.

**Reply by jbr** [reply]
If you run cargo update, you should be able to set the host to a file path, either through the `HOST` env var or `Config::with_host`. Let me know if this works as you'd expect. This was released in [trillium-tokio@0.1.4](https://github.com/trillium-rs/trillium/releases/tag/trillium-tokio-v0.1.4), [trillium-async-std@0.1.4](https://github.com/trillium-rs/trillium/releases/tag/trillium-async-std-v0.1.4), and [trillium-smol@0.1.4](https://github.com/trillium-rs/trillium/releases/tag/trillium-smol-v0.1.4)
